content/docs/reference/programming-model.md

```diff
@@ -115,27 +115,36 @@ The `args` provided to a resource determine what inputs will be used to initiali
 
 All resource constructors also accept an `options` argument which can provide the following additional resource options controlling how the resource will be managed by Pulumi.
 
+###### `additionalSecretOutputs`
+Provides a list of output properties which should be treated as secrets. This value augments any values that Pulumi detects itself, based on what secret inputs to the resource has. This is typically used to express that for a specific instance of a resource, some of its output properties should be treated as secrets (when they would not normally be).
+
+###### `aliases`
+Lets you rename a resource. When making a breaking change to the name or type of a resource or component, you can add the old name to the list of `aliases` for a resource to ensure that existing resources will be migrated to the new name instead of being deleted and replaced with the new named resource.
+
+###### `customTimeouts`
+Lets you specify a request timeout in string format when creating, updating, or deleting a resource. For example, `customTimeOuts: { create: "1m" }`.
+
+###### `deleteBeforeReplace`
+Set this option to `true` to specify that replacements of the resource will delete the existing resource before creating its replacement.  This will lead to downtime during the replacement, but may be necessary for some resources that manage scarce resources behind the scenes.  The default is `false`.
+
 ###### `dependsOn`
 Provides a list of explicit resource dependencies to add to the resource. Every resource referenced either directly or indirectly by an `Output` that is passed in to the resource constructor will implicitly be included, so this additional information is only needed when the dependency is on something that is not already an input to the resource. The default is `[]`.
 
-###### `protect`
-Marks a resource as protected. A protected resource cannot be deleted directly: first you must set `protect: false` and run `pulumi up`. Then, the resource can be deleted, either by removing the line of code or by running `pulumi destroy`.  The default is to inherit this value from the parent resource, and `false` for resources without a parent.
+###### `ignoreChanges`
+Provides a list of properties which will be ignored as part of updates. The value of the property will be used for newly created resources, but will not be used as part of updates. This is typically used to avoid changes in properties leading to diffs or to change defaults for a property without forcing all existing deployed stacks to update or replace the affected resource.
+
+###### `import`
```

Comment:
I specifically changed to `instantiating a Pulumi resource`, but not 100% sure that's accurate. I also removed the `i.e` sections from the change log, but want to make sure this description is still accurate. Thoughts, @pgavlin ?

Also, I was wondering if [this comment](https://github.com/pulumi/pulumi/pull/2893#issuecomment-511468099) needs to be called out.

Reply:
Here's what I wrote:

> The ID of an existing resource to import for management using Pulumi. If this value is set, Pulumi will read the current state of the resource with the given ID from the backing provider (e.g. AWS, Azure, GCP, Kubernetes). The inputs to the resource's constructor must not differ from this state or the import will fail. Once a resource has been imported, this property should be unset.